man/adoc/bpftrace.adoc

```diff
@@ -3804,6 +3808,169 @@ interval:s:10 {
 
 Set all values for all keys to zero.
 
+[#map-functions-tseries]
+=== tseries
+
+.variants
+* `tseries_t tseries(int64 n, int64 interval_ns, int64 num_intervals)`
+* `tseries_t tseries(int64 n, int64 interval_ns, int64 num_intervals, const string agg)`
+
+Create a time series that tracks an integer value. `tseries` records up to
+`num_intervals` intervals representing `interval_ns` nanoseconds.
+
+==== Durations
+
+`interval_ns` is an unsigned integer that specifies the interval duration. You
+may use numbers with duration suffixes to improve readability:
+
+----
+@a = tseries(1, 100ns, 5); // 100 nanoseconds
+@b = tseries(1, 100us, 5); // 100 microseconds
+@c = tseries(1, 100ms, 5); // 100 milliseconds
+@d = tseries(1, 1s, 5);    // 1 second
+----
+
+==== Aggregation Functions
+
+By default, each interval in `tseries` contains the last value recorded in that
+interval. The optional `agg` parameter specifies how values in the same interval
+are aggregated.
+
+|===
+| Aggregation Function | Example | Description
+
+| `avg`
+| `@ = tseries(@v, 1s, 5, "avg")`
+| Calculate the running average of all the values in each interval.
+
+| `max`
+| `@ = tseries(@v, 1s, 5, "max")`
+| Calculate the maximum of all values in each interval.
+
+| `min`
+| `@ = tseries(@v, 1s, 5, "min")`
+| Calculate the minimum of all values in each interval.
+
+| `sum`
+| `@ = tseries(@v, 1s, 5, "sum")`
+| Calculate the sum of all values in each interval.
+
+|===
+
+==== Examples
+
+Example one:
+
+----
+BEGIN {
+  @i = 0;
+}
+
+// Record the minimum of ten random values generated during each 100ms interval.
+i:ms:10 {
+  @ = tseries(rand % 10, 100ms, 20, "min");
+  @i++;
+
+  if (@i == 200) {
```

Comment:
nit: I don't think we need this `if` block at all.  I would also have the example tseries completely populated below, so maybe just make it 10 `num_intervals`

Reply:
Done